include/stl2/detail/algorithm/includes.hpp

```diff
@@ -21,48 +21,43 @@
 // includes [includes]
 //
 STL2_OPEN_NAMESPACE {
-	template<InputIterator I1, Sentinel<I1> S1,
-		InputIterator I2, Sentinel<I2> S2, class Comp = less,
-		class Proj1 = identity, class Proj2 = identity>
-	requires
-		IndirectStrictWeakOrder<
-			Comp, projected<I1, Proj1>, projected<I2, Proj2>>
-	bool includes(I1 first1, S1 last1, I2 first2, S2 last2, Comp comp = {},
-		Proj1 proj1 = {}, Proj2 proj2 = {})
-	{
-		while (true) {
-			if (first2 == last2) {
-				return true;
+	struct __includes_fn {
+		template<InputIterator I1, Sentinel<I1> S1, InputIterator I2, Sentinel<I2> S2,
+			class Proj1 = identity, class Proj2 = identity,
+			IndirectStrictWeakOrder<projected<I1, Proj1>, projected<I2, Proj2>> Comp = less>
+		constexpr bool
+		operator()(I1 first1, S1 last1, I2 first2, S2 last2, Comp comp = {},
+			Proj1 proj1 = {}, Proj2 proj2 = {}) const
+		{
+			while (true) {
+				if (first2 == last2) {
+					return true;
+				}
+				if (first1 == last1) {
+					return false;
+				}
+				if (__stl2::invoke(comp, __stl2::invoke(proj2, *first2), __stl2::invoke(proj1, *first1))) {
+					return false;
+				}
+				if (!__stl2::invoke(comp, __stl2::invoke(proj1, *first1), __stl2::invoke(proj2, *first2))) {
+					++first2;
+				}
+				++first1;
 			}
-			if (first1 == last1) {
-				return false;
-			}
-			if (__stl2::invoke(comp, __stl2::invoke(proj2, *first2), __stl2::invoke(proj1, *first1))) {
-				return false;
-			}
-			if (!__stl2::invoke(comp, __stl2::invoke(proj1, *first1), __stl2::invoke(proj2, *first2))) {
-				++first2;
-			}
-			++first1;
 		}
-	}
 
-	template<InputRange Rng1, InputRange Rng2, class Comp = less,
-		class Proj1 = identity, class Proj2 = identity>
-	requires
-		IndirectStrictWeakOrder<Comp,
-			projected<iterator_t<Rng1>, Proj1>,
-			projected<iterator_t<Rng2>, Proj2>>
-	bool includes(Rng1&& rng1, Rng2&& rng2, Comp comp = {},
-		Proj1 proj1 = {}, Proj2 proj2 = {})
-	{
-		return __stl2::includes(
-			begin(rng1), end(rng1),
-			begin(rng2), end(rng2),
-			__stl2::ref(comp),
-			__stl2::ref(proj1),
-			__stl2::ref(proj2));
-	}
+		template<InputRange R1, InputRange R2, class Proj1 = identity, class Proj2 = identity,
+			IndirectStrictWeakOrder<projected<iterator_t<R1>, Proj1>,
+				                     projected<iterator_t<R2>, Proj2>> Comp = less>
```

Comment:
This is tabbed (for indentation) and spaced so that it's aligned with the `projected` above, but it looks weird here. Should these spaces be removed?

Reply:
It's weird because there are three tabs on 50 and four tabs on 51. It will align properly if there are the same number of tabs on both lines and a space on the second line for every character in `IndirectStrictWeakOrder<`.

Reply:
Yes, this is annoyingly fiddly, which is why I mostly just add an indent on continuation lines and don't bother with alignment much. If you *really* want them aligned but don't want to bother with spaces, you could add an extra wrap:
```c++
\t\t\tIndirectStrictWeakOrder<
\t\t\t\tprojected</*...*/>,
\t\t\t\tprojected</*...*/>>
```